File: brute-force/BOJ-15649.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var (
	sc       *bufio.Scanner
	wr       *bufio.Writer
	n        int
	m        int
	selected []int
	used     []bool
)

func init() {
	sc = bufio.NewScanner(os.Stdin)
	wr = bufio.NewWriter(os.Stdout)

	// sc.Split(bufio.ScanWords)
}

func main() {
	defer wr.Flush()

	sc.Scan()
	s := strings.Fields(sc.Text())
	n, _ = strconv.Atoi(s[0])
	m, _ = strconv.Atoi(s[1])

	selected = make([]int, m)
	used = make([]bool, n+1)

	recur(0)
}

func recur(k int) {
	if k == m {
		for _, s := range selected {
			fmt.Fprintf(wr, "%d ", s)
		}
		fmt.Fprintln(wr)
	} else {
		for i := 1; i <= n; i++ {
			if used[i] == false {
				selected[k] = i
				used[i] = true
				recur(k + 1)
				used[i] = false
			}
		}
	}
}

File: brute-force/BOJ-15649.py
import sys
n, m = map(int, sys.stdin.readline().split(' '))

selected = [0 for _ in range(m)]
visited = [False for _ in range(n+1)]

def recur(k):
    if k == m:
        for x in selected:
            print(x, end=' ')
        print()
    else:
        for i in range(1, n+1):
            if not visited[i]:
                selected[k] = i
                visited[i] = True
                recur(k+1)
                visited[i] = False

recur(0)